tests/unit/test_charm.py

```diff
@@ -4,72 +4,89 @@
         self.assertEqual(expected_plan, updated_plan)
-        # Check the service was started
-        service = self.harness.model.unit.get_container("httpbin").get_service("httpbin")
+        service = self.harness.model.unit.get_container("mimir").get_service("mimir")
         self.assertTrue(service.is_running())
-        # Ensure we set an ActiveStatus with no message
         self.assertEqual(self.harness.model.unit.status, ActiveStatus())
 
-    def test_config_changed_valid_can_connect(self):
-        # Ensure the simulated Pebble API is reachable
-        self.harness.set_can_connect("httpbin", True)
-        # Trigger a config-changed event with an updated value
-        self.harness.update_config({"log-level": "debug"})
-        # Get the plan now we've run PebbleReady
-        updated_plan = self.harness.get_container_pebble_plan("httpbin").to_dict()
-        updated_env = updated_plan["services"]["httpbin"]["environment"]
-        # Check the config change was effective
-        self.assertEqual(updated_env, {"GUNICORN_CMD_ARGS": "--log-level debug"})
-        self.assertEqual(self.harness.model.unit.status, ActiveStatus())
+    @patch("charm.MimirK8SOperatorCharm._set_alerts", new_callable=Mock)
+    @patch("charm.MimirK8SOperatorCharm._mimir_version", new_callable=PropertyMock)
+    def test_set_alerts_error(self, mock_version, mock_set_alerts):
+        mock_version.return_value = "2.4.0"
+        mock_set_alerts.side_effect = PebbleError
+        self.harness.container_pebble_ready("mimir")
+        self.assertEqual(
+            self.harness.model.unit.status,
+            BlockedStatus("Failed to push updated alert files; see debug logs"),
+        )
 
-    def test_config_changed_valid_cannot_connect(self):
-        # Trigger a config-changed event with an updated value
-        self.harness.update_config({"log-level": "debug"})
-        # Check the charm is in WaitingStatus
-        self.assertIsInstance(self.harness.model.unit.status, WaitingStatus)
+    @patch("charm.MimirK8SOperatorCharm._mimir_version", new_callable=PropertyMock)
+    def test_mimir_pebble_ready_cannot_connect(self, mock_version):
+        mock_version.return_value = "2.4.0"
+        ops.testing.SIMULATE_CAN_CONNECT = False
+        self.harness.container_pebble_ready("mimir")
+        self.assertEqual(self.harness.model.unit.status, WaitingStatus("Waiting for Pebble ready"))
+
+    @patch.object(Container, "push")
+    @patch("charm.MimirK8SOperatorCharm._current_mimir_config", new_callable=PropertyMock)
+    @patch("charm.MimirK8SOperatorCharm._set_alerts", new_callable=Mock)
+    @patch("charm.MimirK8SOperatorCharm._mimir_version", new_callable=PropertyMock)
+    def test_mimir_pebble_ready_cannot_push_config(
+        self, mock_version, mock_set_alerts, mock_current_mimir_config, mock_push
+    ):
+        mock_version.return_value = "2.4.0"
+        mock_set_alerts.return_value = True
```

Comment:
How about `patcher.start()` in setUp():
https://github.com/canonical/prometheus-k8s-operator/blob/7f9c38c6b712c8a4060919c7f46acba9a9ca2117/tests/unit/test_charm_status.py#L45